### PR TITLE
new: remove the channel message if already in that channel when auto chat swapper is invoked

### PR DESCRIPTION
**Pull request by SirNapkin1334**
can fix merge conflicts in the morning if neccessary

### CODE REVIEW

File: src/main/java/club/sk1er/hytilities/handlers/chat/swapper/AutoChatSwapper.java [hunk 1]
```diff
@@ -19,10 +28,38 @@ public void onChatEvent(ClientChatReceivedEvent event) {
...
         }
     }
+
+    public static class ChatChannelSwitchMessageMakerNotter {
+
+        private boolean hasDetected;
+        private final ScheduledFuture<?> f;
+
+        ChatChannelSwitchMessageMakerNotter() { // if the message somehow doesn't send, we unregister after 20 seconds
+            f = Multithreading.schedule(() -> { // to prevent blocking the next time it's used
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/handlers/chat/swapper/AutoChatSwapper.java:43]: better variable name?

File: src/main/java/club/sk1er/hytilities/config/HytilitiesConfig.java [hunk 2]
```diff
@@ -169,6 +169,12 @@
     )
     public static boolean hytilitiesChatSwapper;
 
+    @Property(
+        type = PropertyType.SWITCH, name = "Remove All Chat Message",
+        description = "Hide the \"§aYou are now in the §6ALL§a channel§r\" message when auto-switching.",
+        category = "Chat", subcategory = "Parties")
+    public static boolean hytilitiesHideLeAllChatMessage;
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/config/HytilitiesConfig.java:176]: le? seriously?

File: src/main/java/club/sk1er/hytilities/handlers/chat/whitechat/WhiteChat.java [hunk 3]
```diff
@@ -21,7 +21,9 @@ public void onChatEvent(ClientChatReceivedEvent event) {
 
         Matcher matcher = nonMessage.matcher(event.message.getFormattedText());
 
-        if (matcher.find()) {
+        System.out.println(event.message.getFormattedText());
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/handlers/chat/whitechat/WhiteChat.java:24]: debug

File: src/main/java/club/sk1er/hytilities/handlers/chat/swapper/AutoChatSwapper.java [hunk 4]
```diff
@@ -19,10 +28,38 @@ public void onChatEvent(ClientChatReceivedEvent event) {
...
-        if (statusMatcher.find()) {
-            // todo: maybe remove the line of "you're already in this channel!" when this is ran
-            // but the user is already in all chat, so chat doesn't become a garbled mess
+        if (statusMatcher.matches()) {
+            MinecraftForge.EVENT_BUS.register(new ChatChannelSwitchMessageMakerNotter());
             Minecraft.getMinecraft().thePlayer.sendChatMessage("/chat a");
         }
     }
+
+    public static class ChatChannelSwitchMessageMakerNotter {
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/handlers/chat/swapper/AutoChatSwapper.java:37]: this name makes no sense

**Review by asbyth — CHANGES_REQUESTED** [review]

**Comment by SirNapkin1334** [comment]
bruh it didn't amend properly

**Comment by asbyth** [comment]
conflicts are present, other than that this is fine

**Comment by SirNapkin1334** [comment]
How can I help resolve them?

**Comment by SirNapkin1334** [comment]
nvm it was a giant ass button

File: src/main/java/club/sk1er/hytilities/config/HytilitiesConfig.java [hunk 5]
```diff
@@ -198,6 +198,13 @@
     )
     public static boolean hardcoreHearts;
 
+    @Property(
+        type = PropertyType.SWITCH, name = "Remove All Chat Message",
+        description = "Hide the \"§aYou are now in the §6ALL§a channel§r\" message when auto-switching.",
+        category = "Chat", subcategory = "Parties"
+    )
+    public static boolean hytilitiesHideAllChatMessage;
```
Comment (asbyth) [inline src/main/java/club/sk1er/hytilities/config/HytilitiesConfig.java:206]: hytilities prefix is dead but i can resolve this once merged

**Pull request merged**